src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -874,6 +864,241 @@ TEST_F(TestCASESession, Sigma1ParsingTest)
     TestSigma1Parsing(mem, bufferSize, Sigma1SessionIdTooBig);
 }
 
+TEST_F(TestCASESession, EncodeSigma1Test)
+{
+    CASESession::EncodeSigma1Inputs encodeParams;
+
+    uint8_t random[32];
+    EXPECT_EQ(chip::Crypto::DRBG_get_bytes(&random[0], 32), CHIP_NO_ERROR);
+    encodeParams.initiatorRandom    = ByteSpan(random, sizeof(random));
+    encodeParams.initiatorSessionId = 7315;
+    uint8_t destinationId[32]       = { 0xDE, 0xAD };
+    encodeParams.destinationId      = ByteSpan(destinationId, sizeof(destinationId));
+
+    ReliableMessageProtocolConfig MRPConfig = GetDefaultMRPConfig();
+    encodeParams.initiatorMrpConfig         = &MRPConfig;
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma1 should fail when there is no public key
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma1(msg, encodeParams));
+    }
+
+    Crypto::P256Keypair * EphemeralKey = gDeviceOperationalKeystore.AllocateEphemeralKeypairForCASE();
+    ASSERT_NE(EphemeralKey, nullptr);
+    EXPECT_EQ(CHIP_NO_ERROR, EphemeralKey->Initialize(ECPKeyTarget::ECDH));
+    encodeParams.pEphPubKey = &EphemeralKey->Pubkey();
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma1 will Succeed when Public Key is provided
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma1(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+        // EncodeSigma1 should fail when MRP config is missing
+        encodeParams.initiatorMrpConfig = nullptr;
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma1(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg;
+        // Succeed when MRP Config is provided
+        encodeParams.initiatorMrpConfig = &MRPConfig;
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma1(msg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle nonEmptyMsg = System::PacketBufferHandle::New(100);
+
+        // EncodeSigma1 should fail when the packetBufferHandle passed to it is not empty
+        EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma1(nonEmptyMsg, encodeParams));
+    }
+
+    {
+        System::PacketBufferHandle msg1;
+        System::PacketBufferTLVReader tlvReader;
+        CASESession::ParsedSigma1 parseParams;
+
+        // Round Trip Test: Encode Sigma1, Parse it then verify parsed values
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma1(msg1, encodeParams));
+
+        tlvReader.Init(std::move(msg1));
+        EXPECT_EQ(CHIP_NO_ERROR, ParseSigma1(tlvReader, parseParams));
```

Comment:
ParseSigma1 has side-effects on members, so needs to be done on a new CASESession.